src/python/espressomd/script_interface.pyx

```diff
@@ -89,38 +101,32 @@ cdef class PScriptInterface:
     def _valid_parameters(self):
         return [to_str(p.data()) for p in self.sip.get().valid_parameters()]
 
-    cdef set_sip(self, shared_ptr[ScriptInterfaceBase] sip):
-        self.sip = sip
+    def get_sip(self):
+        """
+        Get pointer to the core object.
+        """
+
+        ret = PObjectRef()
+        ret.sip = self.sip
+        return ret
 
-    def set_sip_via_oid(self, PObjectId id):
-        """Set the shared_ptr to an existing core ScriptInterface object via
-        its object id.
+    cdef set_sip(self, shared_ptr[ObjectHandle] sip):
+        """
+        Set the shared_ptr to an existing core object.
         """
-        oid = id.id
-        try:
-            ptr = get_instance(oid).lock()
-            self.set_sip(ptr)
-        except BaseException:
-            raise Exception("Could not get sip for given_id")
-
-    def id(self):
-        """Return the core class object id (:class:`PObjectId`)."""
-        oid = PObjectId()
-        oid.id = self.sip.get().id()
-        return oid
+
+        self.sip = sip
 
     def call_method(self, method, **kwargs):
         """
         Call a method of the core class.
-
         Parameters
         ----------
```

Comment:
Subtle regression: this newline was required to create two paragraphs, one for the description and one for the "Parameters" title. Without it, the `Parameters` word is appended to the previous paragraph and the dashes trigger an error.

Reply:
Thank you. The error message is misleading then, it complains about the indentation...

Reply:
Yeah that's not a very helpful error message. I thought it came from a Sphinx extension, but it's reproducible with vanilla Sphinx syntax. Looking through the Sphinx issue tracker, the same error message can be raised in different scenarios involving a missing newline character. An open-source project felt the need to document this error message directly [in their user manual](https://openmdao.org/releases/0.0.9/docs/quick-ref/sphinx.html?highlight=unexpected%20indentation#some-basics).